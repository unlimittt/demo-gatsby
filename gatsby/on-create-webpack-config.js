const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
