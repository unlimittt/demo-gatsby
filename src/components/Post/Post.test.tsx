import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Post from './Post';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({adapter: new Adapter()})

describe('Post', () => {
  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(siteMetadata)
    );
  });

  const props = {
    post: {
      id: 'test-123',
      html: '<p>test</p>',
      fields: {
        slug: '/test',
        categorySlug: '/test-category',
        tagSlugs: [
          '/test_0',
          '/test_1'
        ]
      },
      frontmatter: {
        date: '2016-09-01',
        tags: [
          'test_0',
          'test_1'
        ],
        title: 'test'
      }
    }
  };

  it('renders correctly', () => {
    const wrapper = shallow(<Post {...props} />)
    expect(wrapper).toMatchSnapshot()
  });
});
