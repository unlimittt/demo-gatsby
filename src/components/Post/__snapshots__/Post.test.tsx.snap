// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<div
  className="post"
>
  <mockConstructor
    className="post__home-button"
    to="/"
  >
    All Articles
  </mockConstructor>
  <div
    className="post__content"
  >
    <Content
      body="<p>test</p>"
      title="test"
    />
  </div>
  <div
    className="post__footer"
  >
    <Meta
      date="2016-09-01"
    />
    <Tags
      tagSlugs={
        Array [
          "/test_0",
          "/test_1",
        ]
      }
      tags={
        Array [
          "test_0",
          "test_1",
        ]
      }
    />
    <Author />
  </div>
  <div
    className="post__comments"
  >
    <Comments
      postSlug="/test"
      postTitle="test"
    />
  </div>
</div>
`;
